keps/sig-network/1880-services-ip-ranges/README.md

```diff
@@ -0,0 +1,1015 @@
+There is no conflict resolution, each apiserver keep writing and deleting others apiservers allocator bitmap
+and Services.
+
+In order to be completely backwards compatible, the bootstrap process will remain the same, the
+difference is that instead of creating a bitmap based on the flags, it will create new
+ServiceIPRange object from the flags (flags configuration removal is out of scope of this KEP) 
+with a special label `networking.kubernetes.io/service-ip-cidr-default` set to true.
+
+With the new implementation only the Service IP Range configuration will be consistent, but not the
+default IP Family, that will still depend on the flags configuration of each individual apiserver.
+However, users still can provide consistency by using a webhook or admission controller for
+defaulting the IPFamily or IPFamilyPolicy fields on Services (to be revisited before Beta).
+
+There can be multiple default ServiceIPRange objects, also for the same IP family, since
+apiservers with different configuration can work, the only condition is that the defined
+service IP ranges contain the `kubernetes.default` Service.
+
+The new bootstrap process will be:
+
+```
+read_flags
+if invalid flags
+  exit
+get ServiceCIDR objects
+if empty || no default
+   create default ServiceCIDR from flags
+   run
+else
+   if default matches flags
+      run
+   else if overlaps
+      log/event
+      run
+   else
+      log/event periodically
+      run
+```
+
+[![](https://mermaid.ink/img/pako:eNp1ksFuwjAMhl_Fygkk2AP0AFpbBpvEBbYLbQ9W40KkNKnStBuq-u4LLRXZpuUQRf4__7aTdCzXnFjAzgarC7zHqQK3npNQa5vBcrmCt5kh5FCTaUVOy1xwA4XEcz0fZNh3LUrBx9i6h1SNHnsng9JuH9YKTskLWpSZJ1-pHkzC2ZYsHMcS0Wt8qOcjNe7hjf1EYZ8-lBVyTPG1ySfqOBXYSAv0JWq77j2GjNFmoE5Ti5GfG3cl2vwyzeHXj31umxwalXnCMOMKNp1uyUisfiVv_kv25LvFzmvrZyj8G3pY7JLIPZEl_wIzr9yDvA0-Dc8WrCRTouDu-btbLGX2QiWlLHDH-0WmLFW9Q5uKuwobLqw2LChQ1rRg2Fh9vKqcBdY0NEGxQPebyjvVfwMtxq4y)](https://mermaid.live/edit#pako:eNp1ksFuwjAMhl_Fygkk2AP0AFpbBpvEBbYLbQ9W40KkNKnStBuq-u4LLRXZpuUQRf4__7aTdCzXnFjAzgarC7zHqQK3npNQa5vBcrmCt5kh5FCTaUVOy1xwA4XEcz0fZNh3LUrBx9i6h1SNHnsng9JuH9YKTskLWpSZJ1-pHkzC2ZYsHMcS0Wt8qOcjNe7hjf1EYZ8-lBVyTPG1ySfqOBXYSAv0JWq77j2GjNFmoE5Ti5GfG3cl2vwyzeHXj31umxwalXnCMOMKNp1uyUisfiVv_kv25LvFzmvrZyj8G3pY7JLIPZEl_wIzr9yDvA0-Dc8WrCRTouDu-btbLGX2QiWlLHDH-0WmLFW9Q5uKuwobLqw2LChQ1rRg2Fh9vKqcBdY0NEGxQPebyjvVfwMtxq4y)
+
+#### The special "default" ServiceIPRanges
+
+There has to be always one default ServiceIPRange containing the `kubernetes.default` Service.
+
+If the cluster is dual-stack, there has to be also one default ServiceIPRange per IP family.
+
+In order to ensure the flags are covered by a range, the controller will handle the case of the
+ServiceIPRange being deleted forcefully by recreating it. We allow overlapping ranges for this
+reason, and the default ServiceIPRange is denoted by a wellknown label, not a special name.
+
+The controller will also handle the case where the well-known label is removed, re-adding it.
```

Comment:
I thought I commented on this - I don't think we can re-add it if we don't KNOW this range was created by us, and the way we know is that label. So if the user removes that label, all we can do is make a new range.